src/components/frame/LandingPage/hooks.ts

```diff
@@ -0,0 +1,123 @@
+import { useNavigate } from 'react-router-dom';
+
+import { useAppContext } from '~context/AppContext/AppContext.ts';
+import { useGetMembersCountQuery } from '~gql';
+import useBaseUrl from '~hooks/useBaseUrl.ts';
+import { CREATE_COLONY_ROUTE_BASE } from '~routes';
+import { getLastWallet } from '~utils/autoLogin.ts';
+
+export const useLandingPage = () => {
+  const {
+    canInteract,
+    joinedColonies,
+    user,
+    connectWallet,
+    wallet,
+    walletConnecting,
+    userLoading,
+  } = useAppContext();
+
+  const navigate = useNavigate();
+
+  const {
+    data: contributorsCount,
+    fetchMore,
+    loading: contributorsCountLoading,
+  } = useGetMembersCountQuery({
+    variables: {
+      filter: {
+        or: [{ hasReputation: { eq: true } }, { hasPermissions: { eq: true } }],
+      },
```

Comment:
This will be fetching all the colonies in the database. We can filter the query here rather than later in the hook.

This works, but there is maybe a better way to do it:
```
      filter: {
        or: joinedColonies.map((colony) => ({
          and: [
            { colonyAddress: { eq: colony.colonyAddress } },
            {
              or: [
                { hasPermissions: { eq: true } },
                { hasReputation: { eq: true } },
              ],
            },
          ],
        })),
      },
```